pkg/server/storage_api/engine_test.go

```diff
@@ -57,3 +68,33 @@ func TestStatusEngineStatsJson(t *testing.T) {
 		t.Fatal(errors.Errorf("expected engine metrics to be correctly formatted, got:\n %s", formattedStats))
 	}
 }
+
+func TestStatusEngineStatsJsonWithoutTenantCapability(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	srv := serverutils.StartServerOnly(t, base.TestServerArgs{
+		// Note: We're only testing external-process mode because shared service
+		// mode tenants have all capabilities.
+		DefaultTestTenant: base.ExternalTestTenantAlwaysEnabled,
+	})
+	defer srv.Stopper().Stop(context.Background())
+
+	s := srv.ApplicationLayer()
+
+	var engineStats serverpb.EngineStatsResponse
+	// Using SucceedsSoon because we have seen in the wild that
+	// occasionally requests don't go through with error "transport:
+	// error while dialing: connection interrupted (did the remote node
+	// shut down or are there networking issues?)"
+	testutils.SucceedsSoon(t, func() error {
+		actualErr := srvtestutils.GetStatusJSONProto(s, "enginestats/local", &engineStats)
+		// if actual err is nil, then we have a problem.
+		require.Error(t, actualErr) // this line ensures we got an error
```

Comment:
nit: The comments on these two lines are not needed.

Reply:
fixed, thanks!